api/shared/src/main/scala/dagr/api/models/TaskInfo.scala

```diff
@@ -0,0 +1,153 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package dagr.api.models
+
+import dagr.api.DagrApi.TaskId
+import upickle.Js
+import upickle.default.{Reader, Writer}
+
+import scala.collection.mutable.ListBuffer
+import java.time.Instant
+
+/** The execution information for a task.  Used for extrenal read-only access to [[TaskInfo]]. Any execution system
+  * should extend [[TaskInfo]] instead class to store their specific metadata. */
+trait TaskInfoLike  {
+  def id         : Option[TaskId]
+  def attempts   : Int
+  def script     : Option[String]
```

Comment:
`script` and `log` must be a string since `java.nio.file.Path` is not available for `scala-js`.